### PR TITLE
⚠️ Allow passing a custom webhook server

### PR DESCRIPTION
**Pull request by alvaroaleman**
Currently it is impossible to pass a custom webhook server, because we reference the concrete type rather than an interface. Change this to an interface.

### CODE REVIEW

File: pkg/manager/internal.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ type controllerManager struct {
 	// election was configured.
 	elected chan struct{}
 
-	webhookServer *webhook.Server
+	webhookServer webhook.Server
```
Comment (vincepri) [inline pkg/manager/internal.go:133]: What are some use cases that require an interface instead?

Comment (alvaroaleman) [reply pkg/manager/internal.go:133]: Basically anything where the default implementation isn't working. The case I have at hand is that the same binary also generates the certificates, so `Start` has to wait until they exist.

File: pkg/webhook/server.go [hunk 2]
```diff
@@ -79,6 +101,18 @@ type Server struct {
 
 	// WebhookMux is the multiplexer that handles different webhooks.
 	WebhookMux *http.ServeMux
+}
+
+// New constructs a new Server from the provided options.
+func New(o Options) Server {
```
Comment (sbueringer) [inline pkg/webhook/server.go:107]: Should this be called NewServer to end up with webhook.NewServer?

File: pkg/webhook/server.go [hunk 3]
```diff
@@ -79,6 +101,18 @@ type Server struct {
...
+	return &DefaultServer{
+		Options: o,
+	}
+}
+
+// DefaultServer is the default implementation used for Server.
+type DefaultServer struct {
+	Options Options
```
Comment (sbueringer) [inline pkg/webhook/server.go:115]: Wondering if Options should be private given that we also have a constructor (but no strong opinion)

Comment (alvaroaleman) [reply pkg/webhook/server.go:115]: The reason I left it public is that there are a bunch of tests in the `Manager` that check some values in there

File: pkg/webhook/server.go (hunk 2 above)
Comment (alvaroaleman) [reply pkg/webhook/server.go]: Good idea, renamed it

**Comment by sbueringer** [comment]
/lgtm

/hold
not sure if Vince wants to take another look

**Pull request merged**